### PR TITLE
add hedges and maze park

### PR DESCRIPTION
**Pull request by curstwist**
#### Summary

SUMMARY: Content "Add some new hedges and supporting mapgen"
#### Purpose of change
Add new landscaping options.

#### Describe the solution

Add tall and short hedges, some topiary and a gazing ball for outdoor landscaping,  Add 2 parks to use the new furniture.

#### Testing

spawned in game.

#### Additional context

hedge maze 1
hedge maze2

### CODE REVIEW

File: data/json/furniture_and_terrain/furniture-decorative.json [hunk 1]
```diff
@@ -152,6 +152,28 @@
       "items": [ { "item": "rock", "count": [ 1, 6 ] }, { "item": "rebar", "count": [ 1, 2 ] } ]
     }
   },
+  {
+    "type": "furniture",
+    "id": "f_gazing_ball",
+    "name": "gazng ball",
+    "description": "A mirrored glass globe mounted on a cement pedastal.  It warps your reflection when you look into it",
```
Comment (Night-Pryanik) [inline data/json/furniture_and_terrain/furniture-decorative.json:159]:
```suggestion
    "description": "A mirrored glass globe mounted on a cement pedestal.  It warps your reflection when you look into it.",
```

File: data/json/furniture_and_terrain/furniture-decorative.json [hunk 2]
```diff
@@ -152,6 +152,28 @@
...
+    "symbol": "o",
+    "color": "white",
+    "move_cost_mod": -1,
+    "required_str": 10,
+    "flags": [ "PLACE_ITEM", "BLOCKSDOOR", "MINEABLE" ],
```
Comment (Night-Pryanik) [inline data/json/furniture_and_terrain/furniture-decorative.json:164]: Does PLACE_ITEM flag really fits here? Based on the description, there is no place for items there.

File: data/json/furniture_and_terrain/furniture-flora.json [hunk 3]
```diff
@@ -368,5 +368,107 @@
       "ter_set": "t_dirt",
       "items": [ { "item": "stick_long", "count": [ 3, 10 ] }, { "item": "splinter", "count": [ 10, 25 ] } ]
     }
+  },
+  {
+    "type": "furniture",
+    "id": "f_hedge_short",
+    "name": "short hedge",
+    "description": "A shrub that with small green leaves, trimmed into a squared shape. ",
```
Comment (Night-Pryanik) [inline data/json/furniture_and_terrain/furniture-flora.json:376]:
```suggestion
    "description": "A shrub with small green leaves, trimmed into a squared shape.",
```

File: data/json/furniture_and_terrain/furniture-flora.json [hunk 4]
```diff
@@ -368,5 +368,107 @@
...
+      "sound": "crunch!",
+      "sound_fail": "whack!",
+      "items": [ { "item": "stick_long", "count": [ 2, 3 ] }, { "item": "splinter", "count": [ 5, 10 ] } ]
+    }
+  },
+  {
+    "type": "furniture",
+    "id": "f_hedge_tall",
+    "name": "tall hedge",
+    "description": "A tall shrub with small green leaves, trimmed into a squared shape. You can't see past this thick hedgerow",
```
Comment (Night-Pryanik) [inline data/json/furniture_and_terrain/furniture-flora.json:397]:
```suggestion
    "description": "A tall shrub with small green leaves, trimmed into a squared shape.  You can't see past this thick hedgerow.",
```

File: data/json/furniture_and_terrain/furniture-decorative.json [hunk 5]
```diff
@@ -152,6 +152,28 @@
       "items": [ { "item": "rock", "count": [ 1, 6 ] }, { "item": "rebar", "count": [ 1, 2 ] } ]
     }
   },
+  {
+    "type": "furniture",
+    "id": "f_gazing_ball",
+    "name": "gazng ball",
```
Comment (CountAlex) [inline data/json/furniture_and_terrain/furniture-decorative.json:158]: `gazng` -> `gazing`

**Pull request merged**